app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,68 @@
+class TweetController < ApplicationController
+	#==================== NEW ===============================
+	get '/tweets/new' do
+		if !logged_in?
+			redirect to "/login"
+		else
+			erb :'/tweets/create_tweet'
+		end
+	end
+
+	post '/tweets' do
+		if !params[:content].empty?
+			@user = current_user
+			@user.tweets.build(params).save
+			redirect to "tweets"
+		else
+			redirect to "tweets/new"
+		end
+	end
+	#--------------------------------------------------------
+
+
+	#==================== SHOW ==============================
+	get '/tweets/:tweetid' do
+		if logged_in? && tweet_exists?(params[:tweetid])
+			@tweet = Tweet.find_by(params[:tweetid])
+			erb :'/tweets/show_tweet'
+		elsif (logged_in? && !tweet_exists?(params[:tweetid]))
+			erb :'/tweets/error'
+		else
+			redirect to "/login"
+		end
+	end
+	#--------------------------------------------------------
+
+
+	#==================== EDIT ==============================
+	get '/tweets/:tweetid/edit' do
+		if logged_in?
+			@tweet = Tweet.find(params[:tweetid])
```

Comment:
Shouldn't you verify a tweet belongs to the `current_user` (and not just that any user is logged in) before allowing them to edit?